### ISSUE TITLE
Numpy extension API version problem

### ISSUE THREAD

**Issue by krystophny** [redmod-team/profit#75]
When running `profit fit`:
```
RuntimeError: module compiled against API version 0xe but this version of numpy is 0xd
[...]
ImportError: numpy.core.multiarray failed to import
```

This error occurs only if another installed package (e.g. `tensorflow`) requires an older `numpy` version.

**Comment by krystophny** [comment]
For now we should make --nofortran the default because of this. @Rykath ?

**Comment by mkendler** [comment]
This seems to be an issue of pip: pypa/pip#9542. First, the latest numpy version is installed and with this the fortran files are compiled. Then it recognises that another numpy version is necessery (e.g. for tensorflow) and installs this version afterwards.

**Comment by Rykath** [comment]
Thanks @mkendler !
Changing the build requirement to `oldest-supported-numpy` seems to fix the problem with python 3.8. Unfortunately this solution throws an error with python 3.7 (probably because the oldest numpy for 3.7 is "too" old).

We can change the default to not compile fortran and have an extra `--fortran` flag if that is what you want.

**Comment by krystophny** [comment]
There's some news from the numpy mailing-list how to do versioning of numpy correctly. Maybe it helps:

> Hi all,

> We now have a guide for downstream package authors, talking about API and ABI stability, NumPy versioning, testing against NumPy master, and how to add build time and runtime dependencies for numpy: https://numpy.org/devdocs/user/depending_on_numpy.html
> 
> Especially the version constraints and setting upper bounds for install_requires correctly is important - almost no packages do this correctly (or at all really). If your package depends on NumPy and you deal with packaging it, please check it out!
> 
> And for even more practical details on release process steps for a downstream package, see http://scipy.github.io/devdocs/dev/core-dev/index.html#updating-upper-bounds-of-dependencies
> 
> Cheers,
> Ralf
